.github/workflows/rust.yml

```diff
@@ -10,12 +10,11 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  linux:
+  fmt-clippy:
+    name: format and clippy check
```

Comment:
I get making this a separate step, but won't it increase the time CI takes quite a lot? As far as I understand it, as we have it now clippy will build it and what it has already built then won't be rebuild when building the actual binary.
However with this change won't it build for clippy, and then build from scratch again for each OS? Seems like this increases the times we build the whole thing by 1.

I'm probably wrong, I don't know that much about Github Actions, but I would love to be explained why :hugs: